### PR TITLE
Governance file + Remove outdated info

### PR DESCRIPTION
**Pull request by lucasponce**
Added a first governance file with a better maintainers description.

Removed a couple of outdated files:
- ADOPTERS.adoc
- DEMOS.adoc

Note, for demos the idea is to consolidate internal/external demos under the repo:
https://github.com/kiali/demos

### CODE REVIEW

File: GOVERNANCE.md
Comment (abonas) [inline GOVERNANCE.md:21]: I think the term "master" should no longer be used. Probably "main"?

File: GOVERNANCE.md
Comment (abonas) [inline GOVERNANCE.md:22]: I wonder if "good" is the right term here, "good" is subjective. Perhaps something about well structured, clear, maintainable, documented and tested?

File: GOVERNANCE.md
Comment (abonas) [inline GOVERNANCE.md:54]: What is the difference between line 53 and 54?

File: GOVERNANCE.md
Comment (lucasponce) [reply GOVERNANCE.md:21]: Without enter in discussions, for the case of Kiali, the project continues using the "master" branch.
So, I'd add that expression "master branch" to avoid any potential context misunderstanding, but let's not use a different rename of the branch that may cause confusion.

File: GOVERNANCE.md
Comment (lucasponce) [reply GOVERNANCE.md:22]: To me "good" is as good and subjective as "structured, clear, maintainable, documented and tested".
Both expressions are similar and subjective.

File: GOVERNANCE.md
Comment (lucasponce) [reply GOVERNANCE.md:54]: I'm not an expert and I basically re-used the same form used by Jaeger.

My interpretation is:

- Line 53 scope refers to the explicit terms discussed previously.
- Line 54 scope refers to any new thing that is not explicitly mentioned as a guard.

Comment (abonas) [reply GOVERNANCE.md:54]: OK :)

File: GOVERNANCE.md
Comment (abonas) [reply GOVERNANCE.md:21]: I thought that github is renaming master to main for projects, but I guess I misunderstood

Comment (lucasponce) [reply GOVERNANCE.md:21]: The status is that for new repos, github create a "main" branch instead of "master" branch.
For existing projects is untouched.

As a potential migration may have more implications, my suggestion is to use the "master branch" wording just for this doc and with that context.

If in the future we trigger a migration (we have several repos and it's not a transparent process) we can update the doc as well (just to refer to the same branch we use).

File: GOVERNANCE.md
Comment (jotak) [inline GOVERNANCE.md:40]: Question: when the maintainer agrees to leave (e.g. doesn't have time for the job) or is self-resigning, perhaps we don't have to go through a vote?

File: GOVERNANCE.md
Comment (abonas) [reply GOVERNANCE.md:21]: if the branch keeps being named master then the governance should certainly match it. as I said, my understanding around the renaming was not accurate.

File: GOVERNANCE.md
Comment (lucasponce) [reply GOVERNANCE.md:40]: To me that situation may fall on this case as well.
There is some maintainer that would like to drop for whatever reason, then the group comment on that before.

Note, in 99% of the cases these votes don't need to be "formal", just when proposing some action, if someone desagrees and there is no unanimous decision, then it can be setup a formal vote.

I think.

Comment (jotak) [reply GOVERNANCE.md:40]: The way it's written here, it sounds very formal. If we don't like the way it's written, then we should rewrite. In my (very small) experience of OSS with open governance, votes are systematic, even if it always ends up in an unanimous "+1".

Comment (lucasponce) [reply GOVERNANCE.md:40]: I don't understand.
There will be a vote.
Another topic is how that vote is.

An email can work as a vote, or a thumbup or a differente mechanism, this doc shouldn't force about the how, but about the what.

What I understand is that there is a no consensus of that minimum amount of maintainers, an actions shouldn't go there.

How we setup the votes is a different discussion, in my opinion.

Comment (jpkrohling) [reply GOVERNANCE.md:40]: We didn't have the case for a maintainer being removed yet, but I would expect a vote to happen, even if just a symbolic one.

Comment (jotak) [reply GOVERNANCE.md:40]: @jpkrohling yes I think that's the "spirit" of this kind of governance.
Though for Kiali this is a bit weird given the number of maintainers being all from the same team (at least at the moment)

**Comment by lucasponce** [comment]
I've sent a followup commit to address some comments.

Also note that this is a first version and we can discuss, modify and adapt to Kiali's needs on future iterations.

The goal of this PR in combination with the teams update is

- Provide a better structure of Kiali teams (https://github.com/orgs/kiali/teams).
- Provide an initial description of the roles used.

Please, let me know if this is in good status to merge or there is any concernt to address before we got it merged.

Thanks for all the comments.

**Comment by lucasponce** [comment]
Is there any final comment before that we move forward with this file ?

File: GOVERNANCE.md
Comment (jshaughn) [inline GOVERNANCE.md:10]: Although it's OK as is, the wording is a little verbose.  Perhaps:

This privilege is granted with an expectation of responsibility: maintainers care about the Kiali project and want to help it grow and improve.  Above the ability to contribute changes, a maintainer has demonstrated the ability to collaborate well with the team, assign appropriate code reviewers, contribute high-quality code, and be thorough and timely with fixes, tests and documentation.

File: GOVERNANCE.md
Comment (jshaughn) [reply GOVERNANCE.md:22]: +1 I'd change to, "the ability to write high quality code."

File: GOVERNANCE.md
Comment (jshaughn) [inline GOVERNANCE.md:24]: understanding of team policies, processes,  etc.

File: GOVERNANCE.md
Comment (jshaughn) [inline GOVERNANCE.md:25]: project's

File: GOVERNANCE.md
Comment (jshaughn) [inline GOVERNANCE.md:33]: maintainer or committer?

File: GOVERNANCE.md
Comment (jshaughn) [reply GOVERNANCE.md:40]: +1, "Maintainers can be removed by a 2/3 majority vote by maintainers, or by resignation."

**Review by jshaughn — CHANGES_REQUESTED** [review]
This will be a good addition.  A few suggested changes...

**Review by jshaughn — APPROVED** [review]
LGTM!

**Pull request merged**
